pandas/io/tests/test_parsers.py

```diff
@@ -2690,6 +2690,41 @@ def test_uneven_lines_with_usecols(self):
         df = self.read_csv(StringIO(csv), usecols=usecols)
         tm.assert_frame_equal(df, expected)
 
+    def test_read_empty_with_usecols(self):
+        # See gh-12493
+        names = ['Dummy', 'X', 'Dummy_2']
+        usecols = names[1:2]  # ['X']
+
+        def check_usecols(stringIO, expected):
+            df = self.read_csv(stringIO, names=names, usecols=usecols)
+            tm.assert_frame_equal(df, expected)
+
+        # first, check to see that the response of
+        # parser when faced with no provided columns
+        # throws the correct error, with or without usecols
+        errmsg = "No columns to parse from file"
+
+        with tm.assertRaisesRegexp(ValueError, errmsg):
```

Comment:
Okay, that is what I suspected we were heading towards, and I think it will clear up a lot of confusion in the higher up conversations.  FYI that is going to be an API change because we are standardizing across engines.